File: traphe-backend/src/main/java/com/example/traphe_backend/service/SupplierService.java
package com.example.traphe_backend.service;

import com.example.traphe_backend.dto.request.CreateSupplierRequest;
import com.example.traphe_backend.dto.request.UpdateSupplierRequest;
import com.example.traphe_backend.dto.response.SupplierResponse;

import java.util.List;
import java.util.UUID;

public interface SupplierService {

    SupplierResponse createSupplier(CreateSupplierRequest request);

    List<SupplierResponse> getAllSuppliers();

    SupplierResponse updateSupplier(UUID id, UpdateSupplierRequest request);

    void softDeleteSupplier(UUID id);
}

File: traphe-frontend/src/services/supplier.service.ts
import axiosClient from "@/lib/axios-client";
import type { ApiResponse } from "@/types/api.types";

// Supplier response from backend (matches Supplier entity)
export interface SupplierResponse {
  id: string;
  name: string;
  contact_name: string | null;
  phone: string | null;
  address: string | null;
  email: string | null;
  createdAt: string;
  updatedAt: string;
  isDeleted: boolean;
}

// Request to create/update supplier
export interface SupplierRequest {
  name: string;
  contact_name?: string;
  phone?: string;
  address?: string;
  email?: string;
}

export const supplierService = {
  // Get all suppliers
  getAllSuppliers: async () => {
    return axiosClient.get<any, ApiResponse<SupplierResponse[]>>("/admin/suppliers");
  },

  // Get supplier by ID
  getSupplierById: async (id: string) => {
    return axiosClient.get<any, ApiResponse<SupplierResponse>>(
      `/admin/suppliers/${id}`,
    );
  },

  // Create a new supplier
  createSupplier: async (data: SupplierRequest) => {
    return axiosClient.post<any, ApiResponse<SupplierResponse>>(
      "/admin/suppliers",
      data,
    );
  },

  // Update supplier
  updateSupplier: async (id: string, data: SupplierRequest) => {
    return axiosClient.put<any, ApiResponse<SupplierResponse>>(
      `/admin/suppliers/${id}`,
      data,
    );
  },

  // Delete supplier
  deleteSupplier: async (id: string) => {
    return axiosClient.delete<any, ApiResponse<void>>(`/admin/suppliers/${id}`);
  },
};
